### PR TITLE
Add Support for OpenSearch Settings in Setup ConfigOptions List

### PR DESCRIPTION
**Pull request by macoaure**
This pull request introduces support for configuring OpenSearch settings within the Setup ConfigOptions list in ElasticSuite. By adding this feature, users can now specify OpenSearch-specific configurations, enhancing the flexibility and compatibility of ElasticSuite with OpenSearch.

### CODE REVIEW

File: src/module-elasticsuite-core/Setup/ConfigOptionsList.php [hunk 1]
```diff
@@ -38,17 +38,17 @@ class ConfigOptionsList implements ConfigOptionsListInterface
      * Input key for the options
      */
     const INPUT_KEY_ES_HOSTS = 'es-hosts';
-    const INPUT_KEY_ES_SSL   = 'es-enable-ssl';
-    const INPUT_KEY_ES_USER  = 'es-user';
-    const INPUT_KEY_ES_PASS  = 'es-pass';
+    const INPUT_KEY_ES_SSL = 'es-enable-ssl';
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Setup/ConfigOptionsList.php:41]: Don't change this cosmetic, we like things aligned, and changing these lines is not related to your PR.

File: src/module-elasticsuite-core/Setup/ConfigOptionsList.php [hunk 2]
```diff
@@ -38,17 +38,17 @@ class ConfigOptionsList implements ConfigOptionsListInterface
...
+    const INPUT_KEY_ES_PASS = 'es-pass';
 
     /**
      * Path to the values in the deployment config
      */
     const CONFIG_PATH_ES_HOSTS = self::CONF_PREFIX . '/servers';
-    const CONFIG_PATH_ES_SSL   = self::CONF_PREFIX . '/enable_https_mode';
-    const CONFIG_PATH_ES_USER  = self::CONF_PREFIX . '/http_auth_user';
-    const CONFIG_PATH_ES_PASS  = self::CONF_PREFIX . '/http_auth_pwd';
+    const CONFIG_PATH_ES_SSL = self::CONF_PREFIX . '/enable_https_mode';
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Setup/ConfigOptionsList.php:49]: Don't change this cosmetic, we like things aligned, and changing these lines is not related to your PR.

File: src/module-elasticsuite-core/Setup/ConfigOptionsList.php [hunk 3]
```diff
@@ -80,8 +80,8 @@ public function __construct(
         SearchConfigOptionsList $searchConfigOptionsList,
         $fallbackMapping = []
     ) {
-        $this->clientBuilder           = $clientBuilder;
-        $this->fallbackMapping         = array_merge($this->fallbackMapping, $fallbackMapping);
+        $this->clientBuilder = $clientBuilder;
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Setup/ConfigOptionsList.php:83]: Don't change this cosmetic, we like things aligned, and changing these lines is not related to your PR.

**Review by romainruaud — CHANGES_REQUESTED** [review]
Looks good to me except the cosmetic changes in code alignment that I'd like to be removed.

Regards

**Comment by romainruaud** [comment]
@macoaure can you please update your PR ?

Regards

**Comment by macoaure** [comment]
hey @romainruaud sorry for my delay, i have fixed the coding standard.

**Review by rbayet — APPROVED** [review]

**Pull request merged**